configure.ac

```diff
@@ -0,0 +1,74 @@
+AC_INIT([chaosnet-bridge], 0.0)
+AC_CONFIG_AUX_DIR([build-aux])
+AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+
+AC_CONFIG_SRCDIR([cbridge.c])
+
+AC_ARG_ENABLE([etherp],
+    AS_HELP_STRING([--enable-etherp], [Enable Chaos-over-Ether code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-etherp])])],
+    [enable_etherp=yes])
+enable_etherp_val=0
+if test "x$enable_etherp" = "xyes"; then
+    enable_etherp_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_ETHERP], [$enable_etherp_val], [])
+AM_CONDITIONAL([ENABLE_ETHERP], [test x$enable_etherp = xyes])
+
+AC_ARG_ENABLE([tls],
+    AS_HELP_STRING([--enable-tls], [Enable Chaos-over-TLS code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-tls])])],
+    [enable_tls=yes])
+enable_tls_val=0
+if test "x$enable_tls" = "xyes"; then
+    enable_tls_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_TLS], [$enable_tls_val], [])
+AM_CONDITIONAL([ENABLE_TLS], [test x$enable_tls = xyes])
+
+AC_ARG_ENABLE([ip],
+    AS_HELP_STRING([--enable-ip], [Enable Chaos-over-IP code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-ip])])],
+    [enable_ip=yes])
+enable_ip_val=0
+if test "x$enable_ip" = "xyes"; then
+    enable_ip_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_IP], [$enable_ip_val], [])
+AM_CONDITIONAL([ENABLE_IP], [test x$enable_ip = xyes])
+
+# CHAOS_DNS // enable the DNS code (forwarder and internal host/addr parser)
+AC_ARG_ENABLE([dns],
+    AS_HELP_STRING([--enable-dns], [Enable DNS code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-dns])])],
+    [enable_dns=yes])
+enable_dns_val=0
+if test "x$enable_enable_dns" = "xyes"; then
+    enable_dns_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_DNS], [$enable_dns_val], [])
+AM_CONDITIONAL([ENABLE_DNS], [test x$enable_dns = xyes])
+
+#	CHAOS_DNS_SERVER
+#	CHAOS_ADDR_DOMAIN
+
+AC_PROG_CC
+AC_CHECK_HEADER([net/ethertypes.h],
+    [AC_DEFINE([HAVE_NET_ETHERTYPES_H], [1], 
+               [Define to 1 if you have <net/ethertypes.h>])])
+AC_CHECK_HEADER([net/ethernet.h], 
+     [AC_DEFINE([HAVE_NET_ETHERNET_H], [1], 
+                [Define to 1 if you have <net/ethernet.h>.])])
+AC_CHECK_HEADER([netpacket/packet.h],
+    [AC_DEFINE([HAVE_NETPACKET_PACKET_H], [1], 
+               [Define to 1 if you have <netpacket/packet.h>])])
+AC_HEADER_RESOLV
```

Comment:
If this is about resolv.h etc, it's only needed if DNS is enabled? (But I don't know what AC_HEADER_RESOLV does.)

Reply:
AC_HEADER_RESOLV just checks if the header is available, and then sets some variables one can use later, here we also need to check for -lresolv since that is not entirely the same thing as.  When that is done, we can conditionally add the DNS related code.

Reply:
Infact, it is mostly equivalent to:

```
AC_CHECK_HEADER([foo.h],
    [AC_DEFINE([HAVE_FOO_H], [1], 
               [Define to 1 if you have <foo.h>])])
```

Reply:
To quote the manual (https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Particular-Headers.html):

```
— Macro: AC_HEADER_RESOLV

    Checks for header resolv.h, checking for prerequisites first. To properly use resolv.h, your code should contain something like the following:

         #ifdef HAVE_SYS_TYPES_H
         #  include <sys/types.h>
         #endif
         #ifdef HAVE_NETINET_IN_H
         #  include <netinet/in.h>   /* inet_ functions / structs */
         #endif
         #ifdef HAVE_ARPA_NAMESER_H
         #  include <arpa/nameser.h> /* DNS HEADER struct */
         #endif
         #ifdef HAVE_NETDB_H
         #  include <netdb.h>
         #endif
         #include <resolv.h>
```

I don't suggest using that complicated `#ifdef` sequence, since I doubt we need it ever...  But I need to check for <resolv.h> in either case, since that is a requirement for `-lresolv`.